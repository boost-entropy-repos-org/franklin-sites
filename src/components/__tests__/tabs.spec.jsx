import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Tabs from '../tabs';

afterEach(cleanup);

const props = {
  tabData: [
    {
      title: <div>Title 1</div>,
      content: 'blah',
      id: 'id1',
    },
    {
      title: 'Title 2',
      content: 'blaher',
      id: 'id2',
    },
    {
      title: 'Title 3',
      content: 'blahest',
      id: 'id3',
    },
  ],
};

describe('Tabs', () => {
  test('should render', () => {
    const { asFragment } = render(<Tabs {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should show corresponding content when 2nd tab title is clicked', () => {
    const { queryAllByTestId } = render(<Tabs {...props} />);
    const title = queryAllByTestId('tab-title');
    fireEvent.click(title[1]);
    const content = queryAllByTestId('tab-content');
    const firstTabClasses = Object.values(content[0].classList);
    const secondTabClasses = Object.values(content[1].classList);
    const thirdTabClasses = Object.values(content[2].classList);
    expect(firstTabClasses).toEqual(['tabs__content']);
    expect(secondTabClasses).toEqual(['tabs__content', 'tabs__content--active']);
    expect(thirdTabClasses).toEqual(['tabs__content']);
  });
});
