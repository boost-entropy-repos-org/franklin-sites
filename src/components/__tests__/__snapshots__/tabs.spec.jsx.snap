// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render 1`] = `
<DocumentFragment>
  <div
    class="tabs"
  >
    <div
      class="tabs__header"
    >
      <div
        class="tabs__header__item tabs__header__item--active"
        data-testid="tab-title"
        role="button"
        tabindex="0"
      >
        <div>
          Title 1
        </div>
      </div>
      <div
        class="tabs__header__item "
        data-testid="tab-title"
        role="button"
        tabindex="0"
      >
        Title 2
      </div>
      <div
        class="tabs__header__item "
        data-testid="tab-title"
        role="button"
        tabindex="0"
      >
        Title 3
      </div>
    </div>
  </div>
  <div>
    <div
      class="tabs__content tabs__content--active"
      data-testid="tab-content"
    >
      blah
    </div>
    <div
      class="tabs__content "
      data-testid="tab-content"
    >
      blaher
    </div>
    <div
      class="tabs__content "
      data-testid="tab-content"
    >
      blahest
    </div>
  </div>
</DocumentFragment>
`;
